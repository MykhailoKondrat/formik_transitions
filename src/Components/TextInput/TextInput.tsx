import React from "react";
import TextField from "@material-ui/core/TextField";
import { MyInput } from "../../interfaces/interfaces";

const TextInput: React.FC<MyInput> = React.memo(
  ({ label, field, meta, ...props }: MyInput) => {
    return (
      <>
        <TextField
          label={label}
          type="text"
          {...field}
          {...props}
          error={meta?.touched && meta?.error ? true : undefined}
          helperText={meta?.touched ? meta?.error : undefined}
        />
      </>
    );
  }
);

export default TextInput;
