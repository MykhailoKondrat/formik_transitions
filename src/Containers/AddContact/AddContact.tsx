import { InitialValues } from "../../interfaces/interfaces";
import { Formik, FormikValues } from "formik";
import * as Yup from "yup";
import React from "react";
import { Grid } from "@material-ui/core";
import AddContactForm from "../../Components/AddContactForm";

const AddContact: React.FC = () => {
  const initialValues: InitialValues = {
    firstName: "",
    workType: "",
    checked: [],
  };
  return (
    <Grid container={true} direction="column">
      <Formik
        initialValues={initialValues}
        validationSchema={Yup.object({
          firstName: Yup.string()
            .min(5, "Must be 5 characters or more")
            .required("Required"),
          workType: Yup.string().required("Required"),
          checked: Yup.array().required("asd"),
        })}
        onSubmit={(values: FormikValues) => {
          alert(JSON.stringify(values, null, 2));
        }}
        component={AddContactForm}
      />
    </Grid>
  );
};

export default AddContact;
