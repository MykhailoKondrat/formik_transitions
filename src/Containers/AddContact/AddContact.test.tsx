import AddContact from "./AddContact";
import React from "react";
import * as Adapter from "../../../setupTests";
import { shallow } from "enzyme";
import TextInput from "../../Components/TextInput/TextInput";
const adapter = Adapter;

describe("<AddContact/> statefull component", () => {
  it("should match Snapshot", function () {
    expect(shallow(<AddContact />)).toMatchSnapshot();
  });
});
