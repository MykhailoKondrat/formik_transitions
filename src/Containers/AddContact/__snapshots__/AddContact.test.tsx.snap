// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddContact/> statefull component should match Snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <Formik
    component={[Function]}
    initialValues={
      Object {
        "firstName": "",
        "workType": "",
      }
    }
    onSubmit={[Function]}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "workType",
          "firstName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "min": true,
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
          "workType": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
            "type": "string",
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
        "type": "object",
      }
    }
  />
</WithStyles(ForwardRef(Grid))>
`;
